### PR TITLE
Set translation to appear by default; select translation based on user's language (#1400)

### PR DESCRIPTION
**Pull request by blms**
## In this PR

Per #1400:
- Revise ITT panel display logic to show translation with image by default, if both available; then transcription with image if translation is not available; and transcription with translation if image is not available
  - Maintain exceptions for edit modes
  - Apply this to both default "checked" template logic and stimulus controller logic for handling all three clicked
- Choose the first selected/default translation based on the current site language

### CODE REVIEW

File: geniza/corpus/templates/corpus/snippets/document_transcription.html [hunk 1]
```diff
@@ -33,12 +33,12 @@
...
         <label for="transcription-on"><svg><use xlink:href="{% static 'img/ui/all/all/transcription-toggle.svg' %}#transcription-toggle" /></svg></label>
+        {# translation displayed by default; disable if no content is available #}
         {# translators: label for checkbox toggle to show the translation panel for a document #}
-        <input type="checkbox" class="toggle" id="translation-on" data-ittpanel-target="toggle" data-action="ittpanel#clickToggle" aria-label="{% translate 'show translation' %}" {% if not document.digital_translations and edit_mode != "translating" %}disabled="true" {% elif edit_mode == "translating" %}checked="true" {% endif %}/>
+        <input type="checkbox" class="toggle" id="translation-on" data-ittpanel-target="toggle" data-action="ittpanel#clickToggle" aria-label="{% translate 'show translation' %}" {% if not document.digital_translations and edit_mode != "translating" %}disabled="true" {% elif edit_mode == "translating" or not document.has_image %}checked="true" {% elif not edit_mode %}checked="true" {% endif %}/>
```
Comment (rlskoeser) [inline geniza/corpus/templates/corpus/snippets/document_transcription.html:41]: Is there a way to put this logic (which panels to display by default) into the view instead of having so much logic embedded in the template?

File: geniza/corpus/models.py [hunk 2]
```diff
@@ -956,6 +956,19 @@ def digital_translations(self):
...
         ).order_by("source")
 
+    @property
+    def default_translation(self):
+        """The first translation footnote that is in the current language, or the first
+        translation footnote ordered alphabetically by source if one is not available
+        in the current language."""
+
+        translations = self.digital_translations()
+        in_language = translations.filter(source__languages__code=get_language())
```
Comment (rlskoeser) [inline geniza/corpus/models.py:966]: IDK if it matters here but I think filtering incurs an additional db query

File: geniza/corpus/models.py [hunk 3]
```diff
@@ -956,6 +956,19 @@ def digital_translations(self):
...
+        if in_language.exists():
+            return in_language.first()
+        else:
+            return translations.first()
+
```
[the remark this answers is not in the document]
Comment (rlskoeser) [reply geniza/corpus/models.py:971]: is there a meaningful sorting logic for translations otherwise? curious which one gets picked as first

**Review by rlskoeser — APPROVED** [review]

File: geniza/corpus/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (blms) [reply geniza/corpus/models.py]: Yes, the default is to order by source, which is ordered alphabetically by title, then by year.

File: geniza/corpus/models.py (hunk 2 above)
Comment (blms) [reply geniza/corpus/models.py:966]: Curiously, it seems to be five queries either way, just with an `INNER JOIN` when filtering. Without that line, and just picking `translations.first()`:

```sql
[20/Jul/2023 14:30:32] DEBUG:django.db.backends::(0.001) SELECT "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" WHERE ("django_content_type"."app_label" = 'corpus' AND "django_content_type"."model" = 'document') LIMIT 21; args=('corpus', 'document')
[20/Jul/2023 14:30:32] DEBUG:django.db.backends::(0.006) SELECT "footnotes_footnote"."id", "footnotes_footnote"."source_id", "footnotes_footnote"."location", "footnotes_footnote"."location_sort", "footnotes_footnote"."doc_relation", "footnotes_footnote"."notes", "footnotes_footnote"."content", "footnotes_footnote"."url", "footnotes_footnote"."content_type_id", "footnotes_footnote"."object_id" FROM "footnotes_footnote" INNER JOIN "footnotes_source" ON ("footnotes_footnote"."source_id" = "footnotes_source"."id") WHERE ("footnotes_footnote"."content_type_id" = 8 AND "footnotes_footnote"."object_id" = 17210 AND "footnotes_footnote"."doc_relation"::text LIKE '%Y%') ORDER BY "footnotes_source"."title" ASC, "footnotes_source"."year" ASC LIMIT 1; args=(8, 17210, '%Y%')
[20/Jul/2023 14:30:32] DEBUG:django.db.backends::(0.000) SELECT "corpus_document"."doc_date_original", "corpus_document"."doc_date_calendar", "corpus_document"."doc_date_standard", "corpus_document"."id", "corpus_document"."image_order_override", "corpus_document"."shelfmark_override", "corpus_document"."description", "corpus_document"."description_en", "corpus_document"."description_he", "corpus_document"."description_ar", "corpus_document"."doctype_id", "corpus_document"."language_note", "corpus_document"."notes", "corpus_document"."created", "corpus_document"."last_modified", "corpus_document"."needs_review", "corpus_document"."old_pgpids", "corpus_document"."status" FROM "corpus_document" WHERE "corpus_document"."id" = 17210 LIMIT 21; args=(17210,)
[20/Jul/2023 14:30:32] DEBUG:django.db.backends::(0.001) SELECT "corpus_textblock"."id", "corpus_textblock"."document_id", "corpus_textblock"."fragment_id", "corpus_textblock"."certain", "corpus_textblock"."selected_images", "corpus_textblock"."region", "corpus_textblock"."multifragment", "corpus_textblock"."order" FROM "corpus_textblock" WHERE "corpus_textblock"."document_id" = 17210 ORDER BY "corpus_textblock"."order" ASC; args=(17210,)
[20/Jul/2023 14:30:32] DEBUG:django.db.backends::(0.001) SELECT "corpus_fragment"."id", "corpus_fragment"."shelfmark", "corpus_fragment"."old_shelfmarks", "corpus_fragment"."collection_id", "corpus_fragment"."url", "corpus_fragment"."iiif_url", "corpus_fragment"."is_multifragment", "corpus_fragment"."notes", "corpus_fragment"."needs_review", "corpus_fragment"."manifest_id", "corpus_fragment"."created", "corpus_fragment"."last_modified" FROM "corpus_fragment" WHERE "corpus_fragment"."id" = 17929 LIMIT 21; args=(17929,)
```

as written:

```sql
[20/Jul/2023 14:31:21] DEBUG:django.db.backends::(0.001) SELECT "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" WHERE ("django_content_type"."app_label" = 'corpus' AND "django_content_type"."model" = 'document') LIMIT 21; args=('corpus', 'document')
[20/Jul/2023 14:31:21] DEBUG:django.db.backends::(0.007) SELECT "footnotes_footnote"."id", "footnotes_footnote"."source_id", "footnotes_footnote"."location", "footnotes_footnote"."location_sort", "footnotes_footnote"."doc_relation", "footnotes_footnote"."notes", "footnotes_footnote"."content", "footnotes_footnote"."url", "footnotes_footnote"."content_type_id", "footnotes_footnote"."object_id" FROM "footnotes_footnote" INNER JOIN "footnotes_source" ON ("footnotes_footnote"."source_id" = "footnotes_source"."id") INNER JOIN "footnotes_source_languages" ON ("footnotes_source"."id" = "footnotes_source_languages"."source_id") INNER JOIN "footnotes_sourcelanguage" ON ("footnotes_source_languages"."sourcelanguage_id" = "footnotes_sourcelanguage"."id") WHERE ("footnotes_footnote"."content_type_id" = 8 AND "footnotes_footnote"."object_id" = 17210 AND "footnotes_footnote"."doc_relation"::text LIKE '%Y%' AND "footnotes_sourcelanguage"."code" = 'en') ORDER BY "footnotes_source"."title" ASC, "footnotes_source"."year" ASC LIMIT 1; args=(8, 17210, '%Y%', 'en')
[20/Jul/2023 14:31:21] DEBUG:django.db.backends::(0.000) SELECT "corpus_document"."doc_date_original", "corpus_document"."doc_date_calendar", "corpus_document"."doc_date_standard", "corpus_document"."id", "corpus_document"."image_order_override", "corpus_document"."shelfmark_override", "corpus_document"."description", "corpus_document"."description_en", "corpus_document"."description_he", "corpus_document"."description_ar", "corpus_document"."doctype_id", "corpus_document"."language_note", "corpus_document"."notes", "corpus_document"."created", "corpus_document"."last_modified", "corpus_document"."needs_review", "corpus_document"."old_pgpids", "corpus_document"."status" FROM "corpus_document" WHERE "corpus_document"."id" = 17210 LIMIT 21; args=(17210,)
[20/Jul/2023 14:31:21] DEBUG:django.db.backends::(0.001) SELECT "corpus_textblock"."id", "corpus_textblock"."document_id", "corpus_textblock"."fragment_id", "corpus_textblock"."certain", "corpus_textblock"."selected_images", "corpus_textblock"."region", "corpus_textblock"."multifragment", "corpus_textblock"."order" FROM "corpus_textblock" WHERE "corpus_textblock"."document_id" = 17210 ORDER BY "corpus_textblock"."order" ASC; args=(17210,)
[20/Jul/2023 14:31:21] DEBUG:django.db.backends::(0.001) SELECT "corpus_fragment"."id", "corpus_fragment"."shelfmark", "corpus_fragment"."old_shelfmarks", "corpus_fragment"."collection_id", "corpus_fragment"."url", "corpus_fragment"."iiif_url", "corpus_fragment"."is_multifragment", "corpus_fragment"."notes", "corpus_fragment"."needs_review", "corpus_fragment"."manifest_id", "corpus_fragment"."created", "corpus_fragment"."last_modified" FROM "corpus_fragment" WHERE "corpus_fragment"."id" = 17929 LIMIT 21; args=(17929,)
```

Comment (rlskoeser) [reply geniza/corpus/models.py:950]: oh, of course, it requires the extra queries to get the data no matter what because we haven't prefetched anything. 

Five queries seems like a lot for this, might be worth adding some prefetching; probably reasonable to add to the `digital_translations` method (or somewhere shared by digital translations and digital editions methods?), since we're probably always going to want source and language information when we pull the translations.

File: geniza/corpus/views.py [hunk 4]
```diff
@@ -318,6 +318,23 @@ def get_context_data(self, **kwargs):
...
+            "transcription"
+            if self.object.has_transcription()
+            and (not self.object.has_translation() or not self.object.has_image())
+            else None,
+        ]
+        disabled = [
+            "images" if not self.object.has_image() else None,
+            "translation" if not self.object.has_translation() else None,
+            "transcription" if not self.object.has_transcription() else None,
+        ]
```
Comment (rlskoeser) [inline geniza/corpus/views.py:336]: ❤️ love seeing this configured in the view, thank you!

Can we avoid calling these `has_` methods twice? My quick thoughts on options: either convert them to cached properties, or checking for which ones are disabled first and then using _that_ to determine the defaults.

Comment (rlskoeser) [reply geniza/corpus/views.py:336]: ok, actually - rereading, the code and thinking about the logic for the defaults (which is a little hard to follow), I think refactoring and basing it on what is / is not disabled should simplify. Instead of that complicated  if, and, or statement, you just need a list of which panels are available (based on which ones are not), list them in the order of priority, and then take the first two. I think that also removes the need for the list comprehension filter you're doing on line 362 to filter out the `None` values.

File: geniza/corpus/views.py [hunk 5]
```diff
@@ -810,6 +829,14 @@ def get_context_data(self, **kwargs):
             # transcription always rtl
             text_direction = "rtl"
 
+        # extended show default/disabled logic from document detail view
```
Comment (rlskoeser) [inline geniza/corpus/views.py:832]: This comment isn't quite clear enough to easily understand what's going on. You're overriding the defaults for transcription based on which thing they are editing?

I think it should be possible to simplify these list comprehensions — hopefully in the main detail view to simplify this code.

File: sitemedia/js/controllers/ittpanel_controller.js [hunk 6]
```diff
@@ -14,6 +14,10 @@ export default class extends Controller {
     connect() {
         document.addEventListener("tahqiq-alert", this.boundAlertHandler);
         document.addEventListener("tahqiq-cancel", this.boundCancelHandler);
+        if (this.isDesktop() && this.transcriptionAndTranslationOpen()) {
+            // if transcription + translation both open, align their contents line-by-line
+            this.alignLines();
```
Comment (rlskoeser) [inline sitemedia/js/controllers/ittpanel_controller.js:19]: 👍 

Do you currently support listening for window resize and aligning ( and un-aligning?) when the page crosses the desktop/mobile boundary? Or could that be on the list somewhere?

File: geniza/corpus/models.py (hunk 2 above)
Comment (blms) [reply geniza/corpus/models.py:950]: I can't seem to reduce the number of queries with prefetching, but maybe I'm doing it wrong? (Definitely not my strong suit!)

```py
    def digital_translations(self):
        """All footnotes for this document where the document relation includes
        digital translation."""

        return (
            self.footnotes.select_related("source")
            .prefetch_related(
                "source__authorship_set__creator",
                "source__languages",
            )
            .filter(doc_relation__contains=Footnote.DIGITAL_TRANSLATION)
            .order_by("source")
        )
```

This now results in 8 queries for `digital_translations()` and 7 queries for `default_translation` afterwards (8 if run without `digital_translations()` first).

File: sitemedia/js/controllers/ittpanel_controller.js (hunk 6 above)
Comment (blms) [reply sitemedia/js/controllers/ittpanel_controller.js:19]: The alignment styles are not applied in mobile, but the code doesn't handle resizing within desktop mode, so added an issue for this! #1431

File: geniza/corpus/models.py (hunk 2 above)
Comment (rlskoeser) [reply geniza/corpus/models.py:950]: ok, did a little experimenting in the console - if you're only filtering on source language, then that's the only thing you'd need to fetch. Something like this: 

`doc.footnotes.all().select_related('source').prefetch_related('source__languages')`

In the console, it looks like two queries to me - one to get the footnotes + select related, and one for the prefetch, which always requires a separate query.

There are some `metadata_prefetch` methods on the `Source` and `Footnote` querysets that we used for the data exports, but it seems like it's more than you need here (unless you do need creator? in which case maybe you can just use that).   I'm not sure if there's a good way to share the footnotes with prefetching between translation and edition methods, or if django does that for you automatically.

**Review by rlskoeser — APPROVED** [review]
refactor looks great!

File: geniza/corpus/models.py (hunk 2 above)
Comment (blms) [reply geniza/corpus/models.py:950]: Curious, I'm still getting more queries by prefetching like that—the one additional prefetch query in the original method, but then also an additional one in `default_translation` after having called it once.

without prefetching:

```sql
>>> d.digital_translations()
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.001) SELECT "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" WHERE ("django_content_type"."app_label" = 'corpus' AND "django_content_type"."model" = 'document') LIMIT 21; args=('corpus', 'document')
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.006) SELECT "footnotes_footnote"."id", "footnotes_footnote"."source_id", "footnotes_footnote"."location", "footnotes_footnote"."location_sort", "footnotes_footnote"."doc_relation", "footnotes_footnote"."notes", "footnotes_footnote"."content", "footnotes_footnote"."url", "footnotes_footnote"."content_type_id", "footnotes_footnote"."object_id" FROM "footnotes_footnote" INNER JOIN "footnotes_source" ON ("footnotes_footnote"."source_id" = "footnotes_source"."id") WHERE ("footnotes_footnote"."content_type_id" = 8 AND "footnotes_footnote"."object_id" = 38281 AND "footnotes_footnote"."doc_relation"::text LIKE '%Y%') ORDER BY "footnotes_source"."title" ASC, "footnotes_source"."year" ASC LIMIT 21; args=(8, 38281, '%Y%')
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.000) SELECT "corpus_document"."doc_date_original", "corpus_document"."doc_date_calendar", "corpus_document"."doc_date_standard", "corpus_document"."id", "corpus_document"."image_order_override", "corpus_document"."shelfmark_override", "corpus_document"."description", "corpus_document"."description_en", "corpus_document"."description_he", "corpus_document"."description_ar", "corpus_document"."doctype_id", "corpus_document"."language_note", "corpus_document"."notes", "corpus_document"."created", "corpus_document"."last_modified", "corpus_document"."needs_review", "corpus_document"."old_pgpids", "corpus_document"."status" FROM "corpus_document" WHERE "corpus_document"."id" = 38281 LIMIT 21; args=(38281,)
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.001) SELECT "corpus_textblock"."id", "corpus_textblock"."document_id", "corpus_textblock"."fragment_id", "corpus_textblock"."certain", "corpus_textblock"."selected_images", "corpus_textblock"."region", "corpus_textblock"."multifragment", "corpus_textblock"."order" FROM "corpus_textblock" WHERE "corpus_textblock"."document_id" = 38281 ORDER BY "corpus_textblock"."order" ASC; args=(38281,)
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.001) SELECT "corpus_fragment"."id", "corpus_fragment"."shelfmark", "corpus_fragment"."old_shelfmarks", "corpus_fragment"."collection_id", "corpus_fragment"."url", "corpus_fragment"."iiif_url", "corpus_fragment"."is_multifragment", "corpus_fragment"."notes", "corpus_fragment"."needs_review", "corpus_fragment"."manifest_id", "corpus_fragment"."created", "corpus_fragment"."last_modified" FROM "corpus_fragment" WHERE "corpus_fragment"."id" = 13396 LIMIT 21; args=(13396,)

>>> d.default_translation
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.004) SELECT "footnotes_footnote"."id", "footnotes_footnote"."source_id", "footnotes_footnote"."location", "footnotes_footnote"."location_sort", "footnotes_footnote"."doc_relation", "footnotes_footnote"."notes", "footnotes_footnote"."content", "footnotes_footnote"."url", "footnotes_footnote"."content_type_id", "footnotes_footnote"."object_id" FROM "footnotes_footnote" INNER JOIN "footnotes_source" ON ("footnotes_footnote"."source_id" = "footnotes_source"."id") INNER JOIN "footnotes_source_languages" ON ("footnotes_source"."id" = "footnotes_source_languages"."source_id") INNER JOIN "footnotes_sourcelanguage" ON ("footnotes_source_languages"."sourcelanguage_id" = "footnotes_sourcelanguage"."id") WHERE ("footnotes_footnote"."content_type_id" = 8 AND "footnotes_footnote"."object_id" = 38281 AND "footnotes_footnote"."doc_relation"::text LIKE '%Y%' AND "footnotes_sourcelanguage"."code" = 'en') ORDER BY "footnotes_source"."title" ASC, "footnotes_source"."year" ASC LIMIT 1; args=(8, 38281, '%Y%', 'en')
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.000) SELECT "corpus_document"."doc_date_original", "corpus_document"."doc_date_calendar", "corpus_document"."doc_date_standard", "corpus_document"."id", "corpus_document"."image_order_override", "corpus_document"."shelfmark_override", "corpus_document"."description", "corpus_document"."description_en", "corpus_document"."description_he", "corpus_document"."description_ar", "corpus_document"."doctype_id", "corpus_document"."language_note", "corpus_document"."notes", "corpus_document"."created", "corpus_document"."last_modified", "corpus_document"."needs_review", "corpus_document"."old_pgpids", "corpus_document"."status" FROM "corpus_document" WHERE "corpus_document"."id" = 38281 LIMIT 21; args=(38281,)
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.000) SELECT "corpus_textblock"."id", "corpus_textblock"."document_id", "corpus_textblock"."fragment_id", "corpus_textblock"."certain", "corpus_textblock"."selected_images", "corpus_textblock"."region", "corpus_textblock"."multifragment", "corpus_textblock"."order" FROM "corpus_textblock" WHERE "corpus_textblock"."document_id" = 38281 ORDER BY "corpus_textblock"."order" ASC; args=(38281,)
[08/Aug/2023 15:19:35] DEBUG:django.db.backends::(0.000) SELECT "corpus_fragment"."id", "corpus_fragment"."shelfmark", "corpus_fragment"."old_shelfmarks", "corpus_fragment"."collection_id", "corpus_fragment"."url", "corpus_fragment"."iiif_url", "corpus_fragment"."is_multifragment", "corpus_fragment"."notes", "corpus_fragment"."needs_review", "corpus_fragment"."manifest_id", "corpus_fragment"."created", "corpus_fragment"."last_modified" FROM "corpus_fragment" WHERE "corpus_fragment"."id" = 13396 LIMIT 21; args=(13396,)
```

with prefecthing:

```sql
>>> d.digital_translations()
[08/Aug/2023 15:15:43] DEBUG:django.db.backends::(0.001) SELECT "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" WHERE ("django_content_type"."app_label" = 'corpus' AND "django_content_type"."model" = 'document') LIMIT 21; args=('corpus', 'document')
[08/Aug/2023 15:15:43] DEBUG:django.db.backends::(0.005) SELECT "footnotes_footnote"."id", "footnotes_footnote"."source_id", "footnotes_footnote"."location", "footnotes_footnote"."location_sort", "footnotes_footnote"."doc_relation", "footnotes_footnote"."notes", "footnotes_footnote"."content", "footnotes_footnote"."url", "footnotes_footnote"."content_type_id", "footnotes_footnote"."object_id", "footnotes_source"."id", "footnotes_source"."title", "footnotes_source"."title_en", "footnotes_source"."title_he", "footnotes_source"."title_ar", "footnotes_source"."year", "footnotes_source"."edition", "footnotes_source"."volume", "footnotes_source"."issue", "footnotes_source"."journal", "footnotes_source"."page_range", "footnotes_source"."publisher", "footnotes_source"."place_published", "footnotes_source"."other_info", "footnotes_source"."source_type_id", "footnotes_source"."url", "footnotes_source"."notes" FROM "footnotes_footnote" INNER JOIN "footnotes_source" ON ("footnotes_footnote"."source_id" = "footnotes_source"."id") WHERE ("footnotes_footnote"."content_type_id" = 8 AND "footnotes_footnote"."object_id" = 38281 AND "footnotes_footnote"."doc_relation"::text LIKE '%Y%') ORDER BY "footnotes_source"."title" ASC, "footnotes_source"."year" ASC LIMIT 21; args=(8, 38281, '%Y%')
[08/Aug/2023 15:15:43] DEBUG:django.db.backends::(0.001) SELECT ("footnotes_source_languages"."source_id") AS "_prefetch_related_val_source_id", "footnotes_sourcelanguage"."id", "footnotes_sourcelanguage"."name", "footnotes_sourcelanguage"."code", "footnotes_sourcelanguage"."direction" FROM "footnotes_sourcelanguage" INNER JOIN "footnotes_source_languages" ON ("footnotes_sourcelanguage"."id" = "footnotes_source_languages"."sourcelanguage_id") WHERE "footnotes_source_languages"."source_id" IN (29); args=(29,)
[08/Aug/2023 15:15:43] DEBUG:django.db.backends::(0.000) SELECT "corpus_document"."doc_date_original", "corpus_document"."doc_date_calendar", "corpus_document"."doc_date_standard", "corpus_document"."id", "corpus_document"."image_order_override", "corpus_document"."shelfmark_override", "corpus_document"."description", "corpus_document"."description_en", "corpus_document"."description_he", "corpus_document"."description_ar", "corpus_document"."doctype_id", "corpus_document"."language_note", "corpus_document"."notes", "corpus_document"."created", "corpus_document"."last_modified", "corpus_document"."needs_review", "corpus_document"."old_pgpids", "corpus_document"."status" FROM "corpus_document" WHERE "corpus_document"."id" = 38281 LIMIT 21; args=(38281,)
[08/Aug/2023 15:15:43] DEBUG:django.db.backends::(0.000) SELECT "corpus_textblock"."id", "corpus_textblock"."document_id", "corpus_textblock"."fragment_id", "corpus_textblock"."certain", "corpus_textblock"."selected_images", "corpus_textblock"."region", "corpus_textblock"."multifragment", "corpus_textblock"."order" FROM "corpus_textblock" WHERE "corpus_textblock"."document_id" = 38281 ORDER BY "corpus_textblock"."order" ASC; args=(38281,)
[08/Aug/2023 15:15:43] DEBUG:django.db.backends::(0.000) SELECT "corpus_fragment"."id", "corpus_fragment"."shelfmark", "corpus_fragment"."old_shelfmarks", "corpus_fragment"."collection_id", "corpus_fragment"."url", "corpus_fragment"."iiif_url", "corpus_fragment"."is_multifragment", "corpus_fragment"."notes", "corpus_fragment"."needs_review", "corpus_fragment"."manifest_id", "corpus_fragment"."created", "corpus_fragment"."last_modified" FROM "corpus_fragment" WHERE "corpus_fragment"."id" = 13396 LIMIT 21; args=(13396,)

>>> d.default_translation
[08/Aug/2023 15:15:58] DEBUG:django.db.backends::(0.004) SELECT "footnotes_footnote"."id", "footnotes_footnote"."source_id", "footnotes_footnote"."location", "footnotes_footnote"."location_sort", "footnotes_footnote"."doc_relation", "footnotes_footnote"."notes", "footnotes_footnote"."content", "footnotes_footnote"."url", "footnotes_footnote"."content_type_id", "footnotes_footnote"."object_id", "footnotes_source"."id", "footnotes_source"."title", "footnotes_source"."title_en", "footnotes_source"."title_he", "footnotes_source"."title_ar", "footnotes_source"."year", "footnotes_source"."edition", "footnotes_source"."volume", "footnotes_source"."issue", "footnotes_source"."journal", "footnotes_source"."page_range", "footnotes_source"."publisher", "footnotes_source"."place_published", "footnotes_source"."other_info", "footnotes_source"."source_type_id", "footnotes_source"."url", "footnotes_source"."notes" FROM "footnotes_footnote" INNER JOIN "footnotes_source" ON ("footnotes_footnote"."source_id" = "footnotes_source"."id") INNER JOIN "footnotes_source_languages" ON ("footnotes_source"."id" = "footnotes_source_languages"."source_id") INNER JOIN "footnotes_sourcelanguage" ON ("footnotes_source_languages"."sourcelanguage_id" = "footnotes_sourcelanguage"."id") WHERE ("footnotes_footnote"."content_type_id" = 8 AND "footnotes_footnote"."object_id" = 38281 AND "footnotes_footnote"."doc_relation"::text LIKE '%Y%' AND "footnotes_sourcelanguage"."code" = 'en') ORDER BY "footnotes_source"."title" ASC, "footnotes_source"."year" ASC LIMIT 1; args=(8, 38281, '%Y%', 'en')
[08/Aug/2023 15:15:58] DEBUG:django.db.backends::(0.000) SELECT ("footnotes_source_languages"."source_id") AS "_prefetch_related_val_source_id", "footnotes_sourcelanguage"."id", "footnotes_sourcelanguage"."name", "footnotes_sourcelanguage"."code", "footnotes_sourcelanguage"."direction" FROM "footnotes_sourcelanguage" INNER JOIN "footnotes_source_languages" ON ("footnotes_sourcelanguage"."id" = "footnotes_source_languages"."sourcelanguage_id") WHERE "footnotes_source_languages"."source_id" IN (29); args=(29,)
[08/Aug/2023 15:15:58] DEBUG:django.db.backends::(0.000) SELECT "corpus_document"."doc_date_original", "corpus_document"."doc_date_calendar", "corpus_document"."doc_date_standard", "corpus_document"."id", "corpus_document"."image_order_override", "corpus_document"."shelfmark_override", "corpus_document"."description", "corpus_document"."description_en", "corpus_document"."description_he", "corpus_document"."description_ar", "corpus_document"."doctype_id", "corpus_document"."language_note", "corpus_document"."notes", "corpus_document"."created", "corpus_document"."last_modified", "corpus_document"."needs_review", "corpus_document"."old_pgpids", "corpus_document"."status" FROM "corpus_document" WHERE "corpus_document"."id" = 38281 LIMIT 21; args=(38281,)
[08/Aug/2023 15:15:58] DEBUG:django.db.backends::(0.000) SELECT "corpus_textblock"."id", "corpus_textblock"."document_id", "corpus_textblock"."fragment_id", "corpus_textblock"."certain", "corpus_textblock"."selected_images", "corpus_textblock"."region", "corpus_textblock"."multifragment", "corpus_textblock"."order" FROM "corpus_textblock" WHERE "corpus_textblock"."document_id" = 38281 ORDER BY "corpus_textblock"."order" ASC; args=(38281,)
[08/Aug/2023 15:15:58] DEBUG:django.db.backends::(0.000) SELECT "corpus_fragment"."id", "corpus_fragment"."shelfmark", "corpus_fragment"."old_shelfmarks", "corpus_fragment"."collection_id", "corpus_fragment"."url", "corpus_fragment"."iiif_url", "corpus_fragment"."is_multifragment", "corpus_fragment"."notes", "corpus_fragment"."needs_review", "corpus_fragment"."manifest_id", "corpus_fragment"."created", "corpus_fragment"."last_modified" FROM "corpus_fragment" WHERE "corpus_fragment"."id" = 13396 LIMIT 21; args=(13396,)
```

Comment (blms) [reply geniza/corpus/models.py:950]: In any case, I think we're talking about differences of one query here, so I'm going to go ahead and merge! I can open a new issue to improve prefetching later if needed.
